crypto/dh/dh_key.c

```diff
@@ -54,6 +57,13 @@ int ossl_dh_compute_key(unsigned char *key, const BIGNUM *pub_key, DH *dh)
         return 0;
     }
 
+#ifdef FIPS_MODULE
+    if (DH_check_pub_key(dh, pub_key, &validate) <= 0) {
```

Comment:
We are doing this via EVP_PKEY_derive_set_peer_ex()

Reply:
Agreed